### DISCUSSION TITLE
Is x-data always necessary or did I find a bug?

### DISCUSSION THREAD

**Discussion by JustSxm** [alpinejs/alpine#4414] [1. Help]
Hello, quick question:

Is x-data always necessary  or is it a bug that the second snippet is not working

```html
<div x-data>
  <button x-on:click="console.log('Sent!')" type="button">Test</button>
</div>
```

```html
<button x-on:click="console.log('Sent!')" type="button">Test</button>
```

**Comment by SimoTod** [comment]
Necessary. It's been asked a few times and it's now mentioned on the documentation on different pages. :)

**Comment by ekwoka** [comment]
x-data is required to initialize a tree (unless you have plugins that do it separately).

There's some other ways to get dataless directives to run, but of course, you don't have data...
